src/main/java/edu/wpi/grip/ui/SocketHandleView.java

```diff
@@ -88,29 +85,20 @@ public SocketHandleView(EventBus eventBus, SocketControlView controlView) {
         // When the user drops the connection onto another socket, add a new connection.
         this.setOnDragDropped(dragEvent -> {
             draggingSocket.ifPresent(other -> {
-                Socket inputSocket, outputSocket;
-
-                // Check which socket was the input and which was the output.  The user can create a connection by
-                // dragging either into the other.
-                if (other.isInputSocket()) {
-                    inputSocket = other.getSocket();
-                    outputSocket = this.controlView.getSocket();
-                } else {
-                    inputSocket = this.controlView.getSocket();
-                    outputSocket = other.getSocket();
+                try {
+                    Connection connection = this.controlView.getSocket().connect(other.getSocket());
+                    eventBus.post(new ConnectionAddedEvent(connection));
+                } catch (IllegalArgumentException e) {
+                    System.out.println("Caught Exception:");
+                    e.printStackTrace(System.out);
                 }
-
-                // Add a new connection for the two sockets
-                @SuppressWarnings("unchecked")
-                Connection connection = new Connection(this.eventBus, outputSocket, inputSocket);
-                eventBus.post(new ConnectionAddedEvent(connection));
             });
         });
 
         // Accept input sockets being dragged onto output sockets and vice versa.
         this.setOnDragOver(dragEvent -> {
             draggingSocket.ifPresent(other -> {
-                if (other.isInputSocket() != this.controlView.isInputSocket()) {
```

Comment:
No, this is a UI thing.  Since you get an onDragOver as soon as you start dragging a socket, we need to make sure that the user has dragged it away to another socket before we actually handle those events.